src/core.jl

```diff
@@ -18,10 +18,17 @@ struct Document{T<:NamedTuple}
 
     function Document(text::AbstractString, metadata::T) where {T}
         check_keys(T)
+        new_text = apply_replacements(text)
 
         # Add an initial and final space to ensure that the last word is recognized
-        # as a word boundary.
-        new_text = " " * apply_replacements(text) * " "
+        # as a word boundary, if necessary.
+        if !startswith(new_text, " ") && !endswith(new_text, " ")
+            new_text = string(" ", new_text, " ")
+        elseif !startswith(new_text, " ")
+            new_text = string(" ", new_text)
+        elseif !endswith(new_text, " ")
+            new_text = string(new_text, " ")
+        end
```

Comment:
```suggestion
        if !startswith(new_text, " ")
            new_text = string(" ", new_text)
        end
        if !endswith(new_text, " ")
            new_text = string(new_text, " ")
        end
```

Reply:
I wanted to avoid allocating the string an extra time, in case it's a long one

Reply:
Ah that's fair